lib/hammer_cli_katello/content_view_version.rb

```diff
@@ -259,6 +259,208 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
+        end
+      end
+
+      def content_view_version
+        resource_content_view_versions.call(:show, 'id' => options['option_id'])
+      end
+
+      def resource_content_view_versions
+        HammerCLIForeman.foreman_resource(:content_view_versions)
+      end
+
+      def repository(id)
+        HammerCLIForeman.foreman_resource(:repositories).call(:show, 'id' => id)
+      end
+
+      def packages(id)
+        HammerCLIForeman.foreman_resource(:packages).call(:index, 'repository_id' => id)['results']
+      end
+
+      def errata(id)
+        HammerCLIForeman.foreman_resource(:errata).call(:index, 'repository_id' => id)['results']
+      end
+
+      def export_json(content_view_version, repositories)
+        json = {
+          "name" => content_view_version['content_view']['name'],
+          "major" => content_view_version['major'],
+          "minor" => content_view_version['minor']
+        }
+
+        json["repositories"] = repositories.collect do |repo|
+          {
+            "repo_id" => repo['id'],
+            "repo_label" => repo['label'],
+            "repo_content_type" => repo['content_type'],
+            "repo_backend_identifier" => repo['backend_identifier'],
+            "repo_relative_path" => repo['relative_path'],
+            "repo_on_disk_path" => "/var/lib/pulp/published/yum/https/repos/#{repo['relative_path']}",
+            "rpm_filenames" => repo['packages'].collect { |package| package['filename'] },
+            "errata_ids" => repo['errata'].collect { |errata| errata['errata_id'] }
+          }
+        end
+
+        json
+      end
+    end
+
+    class ImportCommand < HammerCLIKatello::SingleResourceCommand
+      include HammerCLIForemanTasks::Async
+
+      attr_accessor :export_json, :export_tar_dir, :export_tar_file
+
+      action :import
+      command_name "import"
+
+      success_message _("Content view imported.")
+      failure_message _("Could not import the content view")
+
+      option(
+        '--export-tar',
+        'EXPORT_TAR',
+        _("Location of export tar on disk.")
+      )
+
+      validate_options do
+        option(:option_export_tar).required
+      end
+
+      build_options
+
+      def execute
+        fail unless File.exist?(options['option_export_tar'])
+
+        self.export_tar_file = File.basename(options['option_export_tar'])
+        self.export_tar_dir = File.dirname(options['option_export_tar'])
+        untar_export
+
+        cv = content_view(@export_json['name'])
+        sync_repositories
+        repo_units = @export_json['repositories'].collect { |repo| repo['rpm_filenames'] }.uniq
+        publish(cv['id'], @export_json['major'], @export_json['minor'], repo_units)
+      end
+
+      def untar_export
+        Dir.chdir(@export_tar_dir) do
+          `tar -xf #{@export_tar_file}`
+
+          if File.exist?("#{@export_tar_file.gsub('.tar', '-repos.tar')}")
+            `tar -xf #{@export_tar_file.gsub('.tar', '-repos.tar')}`
```

Comment:
On export it looks like we are getting exported-repos-<id>.tar which causes this to fail:
~~~
exported-repos-3.tar
~~~

~~~
[root@import ~]# ls
anaconda-ks.cfg  export-3.json  export-3.tar  exported-repos-3.tar  ssl-build
hammer> content-view version import --export-tar /root/exported-repos-3.tar
Could not import the content view:
  Error: Export repos tar file is missing
[root@import ~]# mv exported-repos-3.tar exported-repos.tar 
hammer> content-view version import --export-tar /root/exported-repos.tar
Could not import the content view:
  Error: Export repos tar file is missing
hammer> content-view version import --export-tar /root/exported-repos.tar
[root@import ~]# mv exported-repos.tar repos.tar
[root@import ~]# ls
export-3.json  export-3.tar  repos.tar  ssl-build
[root@import ~]# hammer shell
Welcome to the hammer interactive shell.
Type 'help' for usage information.
hammer> content-view version import --export-tar /root/repos.tar
Could not import the content view:
  Error: Export repos tar file is missing
hammer> content-view version import --export-tar /root/export-3.tar
Could not import the content view:
  Error: Export repos tar file is missing
~~~
Tried renaming it to a few things and its not seeing it, selinux is off

Reply:
I tried to clean this up. I think they should be consistent now.

Reply:
this worked for me, :+1: to resolve comment